cmd/ancientremote/server/server.go

```diff
@@ -88,15 +87,6 @@ func MakeServerConfig(c *cli.Context) AncientServerConfig {
 	}
 }
 
-func checkImplementsRemoteFreezerAPI(rpcAPIs []rpc.API) {
```

Comment:
I think this is not business logic in this sense. Server represents a framework. External people that use the framework, have no idea about the dynamic requirements of framework. The framework assumes you register at least one api that implements AncientStore.

I.e. a developer creates an ancient remote server using the framework doesn't register anything that implements a freezer api. The service won't work. This check can't be done by server in tests, because server has no knowledge of what a person is or isn't registering.

The check acts as an affordance for the developer. In fact if you don't want to do things in this way, the idea is the developer doesn't use any of the convenience packages for implementation.

Gist is it's just opinionated :)

Reply:
> External people that use the framework, have no idea about the dynamic requirements of framework. The framework assumes you register at least one api that implements AncientStore.

Let's not use the word "framework." It's pretty ambiguous. 

The server _must_ implement a static (not dynamic) API. Geth needs a clear way of defining and communicating the specification of that API (which, thankfully, is very simple!).  With a clearly defined API, we can -- and should -- then include an integration test that can demonstrate a functioning server as a mock for the built in client. 

If a server fails to properly implement the API, then imo geth's client should just fail loudly and immediately when a method call returns an error. There will be different contexts of errors (transport, logic), and at least to begin I would treat them all as `Fatal`. Eventually if we get sophisticated, maybe transport errors can get some try-again leeway or something from geth's client.